Assets/Scripts/Room/RoomScript.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RoomScript : MonoBehaviour
+{
+    // To be assigned via the Unity editor
+    public List<GameObject> doors;
+
+    private SpawnScript spawner;
+    private List<GameObject> enemies;
+    private int numEnemies = 0;
+
+    // TODO somehow keep track of current list of enemies, not just original... how do we know when an enemy dies?
+
+    private void Awake()
+    {
+        spawner = gameObject.transform.Find("SpawnPoints").gameObject.GetComponent<SpawnScript>();
+        enemies = spawner.SpawnEnemies();
+        numEnemies = enemies.Count;
+        Debug.Log("Number of enemies spawned = " + numEnemies);
+    }
+
+    /* Called when any other collision object enters this Room. Used to detect when the player
+     * enters the room. If there are currently enemies, it will lock all doors.
+     */
+    private void OnCollisionEnter(Collision collision)
+    {
+        if (collision.gameObject.tag == "Player")
+        {
+            // Enemies remaining
+            if (numEnemies != 0)
```

Comment:
Unless im missing it, `numEnemies` is never updated. You need to put `numEnemies = enemies.Count` before this conditional. This also implies that when the enemies die, they are somehow removed from this list, which I'm not sure how we would do that. We have to be careful with what we do in this `OnCollisionEnter` method because it will be constantly firing due to the player's double collision thing.

Reply:
A possible idea may be to have the enemies emit an `DeathEvent` which the room can subscribe to. Then as the enemies die, it removes them from the list.

Reply:
Yep, that's a reasonable solution, we'll just need to make sure only the current room is subscribed to that event.

Reply:
It would all be done in this RoomScript. Since it gets the list of enemies, it's just iterate through them and subscribe to their DeathEvents. Im approving this as is though. You can either continue committing to this PR, or make a new one with any changes you want to make.

Reply:
Sounds good! I've tried doing that but am not having luck. It seems like the events aren't registering. 

Side note: based on my log statements, the good news is that OnTriggerEnter doesn't seem to be constantly firing in RoomScript :)

Reply:
All good now thanks to Jason's help!